CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing
+
+Welcome to Espanso!
+We are very happy to have you and we thank you for considering contributing!
+
+We would like to order the contributions like this:
+
+- Before you start hacking, it is important to **ask first** if your idea or bugfix is in order.
+  Create an issue or come and say hi in the `#dev` channel for fixing bugs and adding features involved with coding logic, and `#documentation` for updating primarly the website. Join us at [the discord][`espanso` discord].
+  We don't bite!.
+
+  It would be sad that you do the effort to clone the project, successfully make the PR, but it wasn't in our plans or there is another PR that is currently adressing that issue.
+
+- After the PR is submitted, the workflows will start to lint, check and test the changes. Please try to stay all green ✅.
+- Most of the time we take some time to respond. Sorry, we are few and there SO much to do here!
+
+## General guidelines and philosophy
+
+This is a list of things we would like to have and mantain across time. Please do your best to abide by.
+
+- We are geared towards a full-rust codebase, but today there is quite a bit of C++ code. We can't get rid of it today, but over time we will prefer to stay in the rust side.
```

Comment:
It would be hard/impractical to remove the C++/Objective-C code for the native modules, and doing it in pure rust would offer little benefits in those areas (as interfacing with native APIs would still require a good deal of `unsafe` code, even when a Rust wrapper exists). When interfacing with OS APIs, it's often more practical to use the main language for the platform (eg. C++ on Windows and Objective-C on macOS), as the tooling around it is more mature :)

Another story is the `espanso-modulo` crate. That's full of spaghetti C++ code I've put together quickly to support the Search/Form UI, and the quality is bad

Reply:
Allright! changed for:

> - We are geared towards a mostly-rust codebase, except on interactions with OS native modules (eg. C++ on Windows and Objective-C on macOS). We decided to stay on the native langauges on each platform, but if it's possible to make a change into rust, submit a PR and we'll see what can we do.